charts/kong/values.yaml

```diff
@@ -45,22 +45,42 @@ image:
   # pullSecrets:
   #   - myRegistrKeySecretName
 
-# Specify Kong admin service configuration
-# Note: It is recommended to not use the Admin API to configure Kong
-# when using Kong as an Ingress Controller.
+# Specify Kong admin service and listener configuration
 admin:
+  # Enable creating a Service for the admin API
+  # Disabling this is recommended for most ingress controller configurations
+  # Enterprise users that wish to use Kong Manager with the controller should enable this
   enabled: false
   # If you want to specify annotations for the admin service, uncomment the following
   # line, add additional or adjust as needed, and remove the curly braces after 'annotations:'.
   annotations: {}
   #  service.beta.kubernetes.io/aws-load-balancer-proxy-protocol: "*"
 
-  # HTTPS traffic on the admin port
-  # if set to false also set readinessProbe and livenessProbe httpGet scheme's to 'HTTP'
-  useTLS: true
-  servicePort: 8444
-  containerPort: 8444
-  # Kong admin service type
+  http:
+    # Enable plaintext HTTP listen for the admin API
+    # Disabling this and using a TLS listen only is recommended for most configuration
+    enabled: false
+    servicePort: 8001
+    containerPort: 8001
+    # Set a nodePort which is available if service type is NodePort
+    # nodePort: 32080
+    # Additional listen parameters, e.g. "reuseport", "backlog=16384"
+    parameters: {}
+
+  tls:
+    # Enable HTTPS listen for the admin API
+    enabled: true
+    servicePort: 8444
+    containerPort: 8444
+    # Set a target port for the TLS port in the admin API service, useful when using TLS
+    # termination on an ELB.
+    # overrideServiceTargetPort: 8000
+    # Set a nodePort which is available if service type is NodePort
+    # nodePort: 32443
+    # Additional listen parameters, e.g. "reuseport", "backlog=16384"
+    parameters:
+    - http2
```

Comment:
Is there a reason for adding `http2` in all blocks?
If copy-pasta only, then please remove `http2` from parameters at all places except proxy.

Reply:
Adding it generally shouldn't break anything, and should be beneficial for loading multiple assets from the Portal and Manager services.

The admin API does have it enabled by default in the latest kong.confs. The Enterprise services do not but I suspect that's an unintended omission; they lack the other new parameters also.

Reply:
Concerns:
1. This path hasn't been tested enough so I'm not sure.
2. Make sure that the property is included in Kong Enterprise releases in the past year. Otherwise, this will be same as `reuseport` problem we ran into recently.

If the above are not an issue, I'm fine with merging this in and reverting if needed due to any other issue that comes up.